persistence-cassandra-3.11/src/main/java/io/stargate/db/cassandra/impl/CassandraPersistence.java

```diff
@@ -309,6 +310,25 @@ public boolean isInSchemaAgreement() {
         <= 1;
   }
 
+  @Override
+  public boolean isInSchemaAgreementWithStorage() {
+    // See comment in isInSchemaAgreement()
+    // Here we also exclude _other_ Stargate nodes (by checking isGossipOnlyMember)
+    InetAddress localAddress = FBUtilities.getBroadcastAddress();
+    return Gossiper.instance.getLiveMembers().stream()
+            .filter(
+                ep -> {
+                  EndpointState epState = Gossiper.instance.getEndpointStateForEndpoint(ep);
+                  return epState != null
+                      && !Gossiper.instance.isDeadState(epState)
+                      && (!Gossiper.instance.isGossipOnlyMember(ep) || localAddress.equals(ep));
+                })
+            .map(Gossiper.instance::getSchemaVersion)
```

Comment:
Why we need this map stage? As far as I can see, the returned `schemaVersion` value is not checked anywhere.
There is only a check on `size <= 1;`.

Reply:
We collect schema UUIDs in a set in the following step to remove duplicates. The size check ensures at most one UUID remains.

Reply:
We could refactor this to use some other technique to achieve the same result. I just followed the pattern of other similar methods that check schema versions.

Reply:
>We collect schema UUIDs in a set in the following step to remove duplicates. The size check ensures at most one UUID remains.
It seems implicit, I am +1 on refactoring that if that's not a big effort.

Reply:
refactored